### ISSUE TITLE
I'd like to contribute ansible-trace under ansible-collections

### ISSUE THREAD

**Issue by mhansen** [ansible-community/community-topics#52]
### Summary

I'd like to contribute https://github.com/mhansen/ansible-trace to ansible-collections.

### Additional Information

Hi there, I've developed https://github.com/mhansen/ansible-trace, a tool for visualising where ansible execution time is spent.

I think this could have broad usefulness to the ansible community, and I wonder if the best way to increase it's reach is by contributing it to Ansible somehow.

I was chatting with @gundalow on https://www.reddit.com/r/ansible/comments/q49h2d/ansibletrace_visualise_execution_time_of_ansible/ and he suggested:

> I'm wondering if this could be released as a standalone collection under https://github.com/ansible-collections/
> u/markhnsn If you'd like to raise an issues here https://github.com/ansible-community/community-topics/issues we can use that to track the steps needed to move the repo over.

The repo now has a GPLv3 license. I imagine there might be some more steps to make this ready to land (just guesses):
- [ ] add tests
- [ ] maybe make sure it works on all versions of Python that Ansible supports

**Comment by mhansen** [comment]
Thanks all. I've converted to a collection (this is heaps easier to
install!)

**Comment by mhansen** [comment]
I see there are bunch of similar-ish plugins under
https://github.com/ansible-collections/community.general:

- community.general.cgroup_memory_recap
- community.general.counter_enabled
- community.general.dense
- community.general.elastic
- community.general.loganalytics
- community.general.opentelemetry
etc...

I don't know if this is just an accident of history that so many logs are
in here, maybe it's better to have a new, more granular repository for new
things.

**Comment by felixfontein** [comment]
> I don't know if this is just an accident of history that so many logs are in here, maybe it's better to have a new, more granular repository for new things.

There are two aspects:
1) When the large Ansible repository was split up into ansible-base (now ansible-core) and a set of collections, most plugins and modules that nobody wanted to take ownership of (in a separate collection) were moved into community.general. That's one reason why a lot of them ended up in there.
2) Some new ones were added since then; community.general is a kind of popular target for new plugins and modules since it is included in the Ansible community package, so if you manage to get your plugin/module added there everyone using that package can use your plugin/module out of the box and docs for your plugin/module will show up on docs.ansible.com.

Having a more specialized collection is definitely a good idea IMO, but it is a bit more effort to get it included into the Ansible package (https://github.com/ansible-collections/ansible-inclusion/) and won't happen until (at least) Ansible 6.0.0 since the inclusion window for Ansible 5 just closed. Obviously folks can also install the collection manually until then (or thereafter if they don't use the Ansible package), so it's not really a big drawback :)

**Comment by mhansen** [comment]
Thanks for the guidance. I'm fairly new to this. Maybe it's enough to just
have the easy ansible-galaxy installation. Or perhaps this is general
enough / there would be benefits from community ownership. Not sure. I'm
fairly happy for now to wait and see, or get advice for whether packages
like this should be best in or out of the community package.

It would be pretty nice to have it included in the bundle out of the box
and have docs on docs.ansible.com if it fits with your vision for that repo
(in particular, would be cool if we could link it from the docs about
profiling execution for discoverability), but otherwise I'm probably fairly
happy with the galaxy distribution, it doesn't seem too hard to install.

**Comment by dmsimard** [comment]
Hi @mhansen and sorry for not getting back to you sooner.

First, thanks for working on ansible-trace and congratulations on getting the collection out on galaxy, I'm happy that it works for you :)

We discussed this particular issue in the community meeting today to see how the community working group might help.
We came up with a number of ideas and questions so before settling on anything, I suggested we reach out to you first in order to ask how things were going, what were your expectations and how we might be able to provide guidance.

With your collection now on galaxy, anyone can install it relatively easily and that's great.
If this is OK with you and you are satisfied, we can stop there but there are other possibilities like:
- Moving the repository to the ansible-collections organization
- Renaming the collection from mhansen.ansible_trace to community.ansible_trace

In terms of benefits, some examples are that being in the ansible-collections organization allows us to provide CI resources so PRs can be tested and collections can be automatically published to ansible galaxy when a new git tag is pushed.

Another question that came up during the meeting was whether it would be appropriate to create a more general purpose collection (say, community.performance_tools or community.debug_tools) but we would need to get traction on it.

Let us know what you think ? Happy to answer questions.

**Comment by mhansen** [comment]
> With your collection now on galaxy, anyone can install it relatively
easily and that's great.
If this is OK with you and you are satisfied, we can stop there

I can confirm I’m satisfied with the galaxy arrangement. I didn’t know
about galaxy at the start of this, and it solves the install and use
problem nicely I think.

I’m very open to making this community or part of ansible collections,
those benefits you list are good. A performance tools collection might also
be good (but tradeoff is it might suffer from many unrelated tools wanting
different release schedules? Not sure).

In conclusion, I’m happy as is, I’m also happy to contribute to something
community or a more core-ish collection, but I think I’ll let you decide if
that’s desired / if there is a broader pattern here. e.g. if you decide
ansible needs a tracer built in and want to make that be ansible-trace or a
fork or rewrite of my code, I’m cool with that. Otherwise fine as is.
Thanks!
